packages/react/src/hooks/NovuProvider.tsx

```diff
@@ -0,0 +1,30 @@
+import { Novu, NovuOptions } from '@novu/js';
+import { createContext, useContext, useMemo } from 'react';
+
+type NovuProviderProps = NovuOptions & {
+  children: JSX.Element;
+};
+
+const NovuContext = createContext<Novu | undefined>(undefined);
+
+export const NovuProvider = (props: NovuProviderProps) => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** I'm not sure what we should do with the props on the Inbox in this case:
```
<NovuProvider subscriberId="subscriberId" applicationIdentifier="applicationIdentifier">
   <Inbox subscriberId="subscriberId" applicationIdentifier="applicationIdentifier" />
</NovuProvider>
```